### PR TITLE
feat: updating string literal to use `"`(double quote) and `'`(single…

### PR DESCRIPTION
**Pull request by vdua**
…-quote) for identifiers

## Related Issue
#64 

## Tasks

- [X] I have signed the [CLA](http://adobe.github.io/cla.html).
- [x] I have written tests and verified that they fail without my change.

### CODE REVIEW

**Comment by vdua** [comment]
A few changes are pending

- update test cases
- the code currently passes the following expression `\"foo`. Fixing that.
- antlr grammar has a duplicate rule `RAW_STRING` and `STRING`. Also some expressions are throwing parsing error via antlr.

File: src/jmespath/Lexer.js [hunk 1]
```diff
@@ -299,7 +299,10 @@ export default class Lexer {
     }
     this._current += 1;
     const literal = stream.slice(start + 1, this._current - 1);
-    return literal.replaceAll("\\'", "'");
+    if (this._current > maxLength) {
```
Comment (vdua) [inline src/jmespath/Lexer.js:302]: Seems like earlier unterminated string literals were supported. i.e "'Something" was valid expression. I assume that was unintentional ?

File: test/jmespath/jmespath.spec.js [hunk 2]
```diff
@@ -75,35 +61,30 @@ function executeTestWithFields(desc, tst) {
...
+  'current', 'escape',
+  'filters', 'functions',
+  'identifiers', 'indices',
+  'literal', 'multiselect',
+  'pipe', 'slice',
+  'syntax', 'unicode',
+  'wildcard'
+];
+
+suites.forEach(suite => {
```
Comment (vdua) [inline test/jmespath/jmespath.spec.js:75]: I find it more readable in the command line and easy to identify which test has failed. I can revert this file, if you find it problematic.

File: antlr/JSONFormula.g4 [hunk 3]
```diff
@@ -34,7 +35,7 @@ expression
   | literal # literalExpression
   | functionExpression # functionCallExpression
   | expression '|' expression # pipeExpression
-  | RAW_STRING # rawStringExpression
+  | (STRING | RAW_STRING) # rawStringExpression
```
Comment (vdua) [inline antlr/JSONFormula.g4:38]: Do we want RAW_STRING and STRING to be separate ? The difference between them is inside JSON literals only a few characters can be escaped while outside all characters are.
```
`"\\z"` is not valid
"\\z" is valid
```

Comment (vdua) [reply antlr/JSONFormula.g4:38]: And the above is true as per antlr grammar. In our Parser, we support both of them.

File: src/jmespath/Lexer.js (hunk 1 above)
Comment (JohnBrinkman) [reply src/jmespath/Lexer.js:302]: Yes, unintentional.  We should be strict

File: test/jmespath/jmespath.spec.js (hunk 2 above)
Comment (JohnBrinkman) [reply test/jmespath/jmespath.spec.js:75]: seems fine to me

File: antlr/JSONFormula.g4 (hunk 3 above)
Comment (JohnBrinkman) [reply antlr/JSONFormula.g4:38]: Hmm.  It appears that not all characters can be escaped?
`JSON.parse('"\\z"')` results in an error. 
Whereas 
`JSON.parse('"\\""')` or `JSON.parse('"\\n"')` succeed.

Comment (JohnBrinkman) [reply antlr/JSONFormula.g4:38]: So maybe our parser should be more strict?

Comment (JohnBrinkman) [reply antlr/JSONFormula.g4:38]: https://datatracker.ietf.org/doc/html/rfc8259#section-7

Comment (vdua) [reply antlr/JSONFormula.g4:38]: https://github.com/adobe/json-formula/issues/70

**Review by JohnBrinkman — COMMENTED** [review]
👍

**Pull request merged**